src/js_printer.zig

```diff
@@ -206,12 +218,28 @@ pub fn quoteForJSONBuffer(text: []const u8, bytes: *MutableString, comptime asci
     const n: usize = text.len;
     while (i < n) {
         const width = strings.wtf8ByteSequenceLengthWithInvalid(text[i]);
-        const c = strings.decodeWTF8RuneT(text.ptr[i .. i + 4][0..4], width, i32, 0);
+        const clamped_width = @min(@as(usize, width), n -| i);
+        const c = strings.decodeWTF8RuneT(
+            &switch (clamped_width) {
+                // 0 is not returned by `wtf8ByteSequenceLengthWithInvalid`
+                1 => .{ text[i], 0, 0, 0 },
+                2 => text[i..][0..2].* ++ .{ 0, 0 },
+                3 => text[i..][0..3].* ++ .{0},
+                4 => text[i..][0..4].*,
+                else => unreachable,
+            },
+            width,
+            i32,
+            0,
+        );
         if (canPrintWithoutEscape(i32, c, ascii_only)) {
             const remain = text[i + @as(usize, width) ..];
```

Comment:
Is it possible this and line 238 can also go out of bounds? I think `width` should be `clamped_width`